alchemist-api/src/main/java/it/unibo/alchemist/model/interfaces/Incarnation.java

```diff
@@ -75,16 +75,13 @@
      *            the random engine
      * @param environment
      *            the environment that will host this object
-     * @param node
-     *            the node that will host this object
      * @param parameter
      *            a {@link String} describing the object
      * @return a new {@link TimeDistribution}
      */
     TimeDistribution<T> createTimeDistribution(
             RandomGenerator randomGenerator,
             Environment<T, P> environment,
```

Comment:
I think it would be better to make the parameter optional (aka: `@Nullable`) rather than removing it. The documented behaviour is: if it's null, it is a reaction not belonging to a `Node`). Otherwise, it is a "normal" reaction.

Reply:
But the method is for the creation of a `TimeDistribution` not a `Reaction`. The node parameter is never used.

Reply:
But it might. In general, a `TimeDistribution` might use information from the `Node` (in which case it'd be unusable for `GlobalReactions`). So the parameter must be there, annotated with `@Nullable`, and should not disappear (even if all the existing implementations do not use it).

Reply:
Alright